test/unit/pipeline/FetchUnitTest.cc

```diff
@@ -101,14 +104,17 @@ TEST_F(PipelineFetchUnitTest, TickStalled) {
 
 // Tests that the fetch unit will handle instructions that straddle fetch block
 // boundaries by automatically requesting the next block of data.
-TEST_F(PipelineFetchUnitTest, FetchUnaligned) {
+TEST_P(PipelineFetchUnitTest, FetchUnaligned) {
   MacroOp mOp = {uopPtr};
   ON_CALL(isa, getMaxInstructionSize()).WillByDefault(Return(insnMaxSizeBytes));
+  ON_CALL(isa, getMinInstructionSize()).WillByDefault(Return(insnMinSizeBytes));
   ON_CALL(memory, getCompletedReads()).WillByDefault(Return(completedReads));
 
-  // Set PC to 14, so there will not be enough data to start decoding
+  assert(insnMinSizeBytes > 1 && "min insn size too small to set PC correctly");
+  uint64_t setPC = (blockSize - insnMinSizeBytes) + 1;
+  // Set PC so that there will not be enough data to start decoding
   EXPECT_CALL(isa, predecode(_, _, _, _)).Times(0);
-  fetchUnit.updatePC(14);
+  fetchUnit.updatePC(setPC);
   fetchUnit.tick();
 
   // Expect a block starting at address 16 to be requested when we fetch again
```

Comment:
Do we knnow that the next block requested will always be 16? Can we not use blockSize instead to be certain?